emperor/support_files/js/sceneplotview3d.js

```diff
@@ -591,21 +591,26 @@ define([
 
   /**
    *
-   * Helper method to remove objects that match a prefix from the view's scene
-   * this method is used by removeAxes and removeAxesLabels. This function
-   * iterates "num" times, and for each iteration it finds and removes objects
-   * with the name of the form "prefix" + "iteration".
+   * Helper method to remove objects with some prefix from the view's scene
    *
-   * @param {String} prefix The label that will prepended to the iterating
-   * index.
+   * @param {String} prefix The prefix of object names to remove
    *
    */
   ScenePlotView3D.prototype._removeObjectsWithPrefix = function(prefix) {
     var scope = this;
-    _.each(_.range(this.decViews.scatter.decomp.dimensions), function(i) {
```

Comment:
After looking into this, I think the old function would still work, as long as we update the objects we iterate over. Instead of iterating over decomp.dimensions (which would go from 1 - N), we need to iterate over scenePlotView.visibleDimensions (which tracks the index of the visible dimensions). Something like this should work:


```javascript
  ScenePlotView3D.prototype._removeObjectsWithPrefix = function(prefix) {
    var scope = this;
    _.each(this.visibleDimensions, function(i) {
      var objectWithPrefix = scope.scene.getObjectByName(prefix + i);
      scope.scene.remove(objectWithPrefix);
    });
  };
```